### PR TITLE
i18n: support for <html lang="$lang">

### LINKED ISSUE #2354: Add <html lang="$some_lang"> to HTML templates
# Feature request

## Description

All pages should have the `<html lang="en-US">` or other appropriate language tag.

Reference: https://www.w3schools.com/Tags/ref_language_codes.asp

## User Stories

As a source/journalist, I want my browser to appropriately handle things like CSS text transforms.

### PR DESCRIPTION
**Pull request by heartsucker**
## Status

Ready for review

## Description of Changes

Fixes #2354.

Adds the RFC 5646 language tag to the HTML.

## Testing

`pytest tests`.

More complete testing depends on merging in some translations.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### CODE REVIEW

**Comment by dachary** [comment]
It looks like [w3school list](https://www.w3schools.com/Tags/ref_language_codes.asp) contradicts the [wikipedia list](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) for ISO-638-1 codes. If we are to believe wikipedia, getting the first part of the locale returned by get_locale() (without the **script** part that only exists in chinese), it would work. What do you think ?

**Comment by heartsucker** [comment]
Yeah, I thought about just truncating it and hoping for the best. Let's see what other people think.

**Comment by dachary** [comment]
For completeness, it can be done with the following babel snippet:
<pre>
from babel import core
core.parse_locale('fr_FR')[0] => fr
core.parse_locale('zh_Hant')[0] => zh
etc.
</pre>
This is essentially what get_locale() does with two useful additions:
* it will accept **-** instead of **_** 
* it will fallback to a supported locale if the requested locale is not availble (i.e. DEFAULT_LOCALE)
So using the following function
<pre>
def get_language():
    return get_locale().split('_')[0]
</pre>
should get you what you need for the lang attribute.

**Comment by heartsucker** [comment]
@dachary Ok, I'm going to use that and then add the exceptions for `zh_Han{s,t}` as they are the two obvious outliers.

**Comment by heartsucker** [comment]
@redshiftzero Can someone restart travis? It looks like a transient OS error.

**Comment by redshiftzero** [comment]
Restarted :-)

**Comment by heartsucker** [comment]
I think we have a new, different transient CircleCI error :(

**Comment by msheiny** [comment]
> I think we have a new, different transient CircleCI error :(

I'm seeing that on my AWS branch --- 👍 and I ended up doing an `XFAIL` which i have to fix. I wasnt able to reproduce in CI manually though. its really annoying. is there a fancy way to re-run that multiple times until it passes with pytest?

File: securedrop/tests/test_i18n.py [hunk 1]
```diff
@@ -173,6 +174,33 @@ def test_verify_default_locale_en_us_if_not_defined_in_config(self):
...
+        assert i18n.locale_to_rfc_5646('en-us') == 'en'
+        assert i18n.locale_to_rfc_5646('zh-hant') == 'zh-Hant'
+
+    def test_html_lang_correct(self):
+        app = journalist.app.test_client()
+        resp = app.get('/', follow_redirects=True)
+        html = resp.data.decode('utf-8')
+        assert re.compile('<html .*lang="en".*>').search(html), html
+
+        # TODO add check for a non-English language on the journalist app
```
Comment (redshiftzero) [inline securedrop/tests/test_i18n.py]: Can you remove these TODOs and add a @checklist item in #2230 to remind us we should add testing for `lang="fr"` when we merge in the French translation?

File: securedrop/i18n.py [hunk 2]
```diff
@@ -129,3 +129,13 @@ def get_locale2name():
...
+
+
+def locale_to_rfc_5646(locale):
+    lower = locale.lower()
+    if 'hant' in lower:
+        return 'zh-Hant'
+    elif 'hans' in lower:
+        return 'zh-Hans'
+    else:
+        return core.parse_locale(locale.split('-')[0])[0]
```
Comment (dachary) [inline securedrop/i18n.py]: <pre>
locale.split('-')[0]
</pre>
should be enough here ?

**Comment by heartsucker** [comment]
@dachary @redshiftzero This is ready for re-review.

File: securedrop/tests/test_i18n.py [hunk 3]
```diff
@@ -211,6 +219,29 @@ def test_verify_default_locale_en_us_if_not_defined_in_config(self):
...
+        resp = app.get('/', follow_redirects=True)
+        html = resp.data.decode('utf-8')
+        assert re.compile('<html .*lang="en".*>').search(html), html
+
+        app = source.app.test_client()
+        resp = app.get('/', follow_redirects=True)
+        html = resp.data.decode('utf-8')
+        assert re.compile('<html .*lang="en".*>').search(html), html
+
+        # check '/generate' too because '/' uses a different template
```
Comment (dachary) [inline securedrop/tests/test_i18n.py]: I don't see how this is different from the above ? did you mean app.get('/generate') ?

**Comment by redshiftzero** [comment]
Once @dachary's [comment here](https://github.com/freedomofpress/securedrop/pull/2355/files/78879c19572f560f9bd78707b9b2e98e7eb93b12#r140652041) is addressed (a good catch), this is ready to merge.

**Comment by heartsucker** [comment]
@redshiftzero @dachary ping

**Comment by heartsucker** [comment]
CircleCI seems to have hiccuped. Restart?

**Comment by redshiftzero** [comment]
Restarted

**Pull request merged**